scripts/sysman/firmware.yml

```diff
@@ -90,6 +90,49 @@ returns:
     - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
         - "User does not have permissions to perform this operation."
 --- #--------------------------------------------------------------------------
+type: function
+desc: "Find Firmware Flash Progress"
+version: "1.7"
+class: $sFirmware
+name: UpdateProgress
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_firmware_handle_t
+      name: hFirmware
+      desc: "[in] Handle for the component."
+    - type: "uint32_t*"
+      name: pCompletionPercent
+      desc: "[in,out] Pointer to the Completion Percentage of Firmware Update"
+returns:
+    - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
+        - "User does not have permissions to perform this operation"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Collect Firmware Logs"
+version: "1.7"
+class: $sFirmware
+name: Logs
```

Comment:
name could be more descriptive about what kind of FW log this is? i.e, if this is an error log, then name could be "GetErrorLog" ?

Reply:
Initial requirement to get AMC FW log. Keeping it generic to expand the scope in future to collect even GPU FW logs

Reply:
> Initial requirement to get AMC FW log. Keeping it generic to expand the scope in future to collect even GPU FW logs

So please share if there should be a parameter to indicate which FW log this is returning ?
Also zesFirmare"Get"Log could be used

Reply:
@vaidyan1-int Query:
For one firmware handle could we retrieve multiple type of logs? If yes then we might need to add some logType parameter in https://spec.oneapi.io/level-zero/latest/sysman/api.html#_CPPv425zes_firmware_properties_t, and accordingly we need to pass on additional argument to API which will tell about type of log.

Reply:
@jitendrasharma1989 yeah that's a good point, we can have API name generic such as FirmwareGetLog and have log Type enum as an input

Reply:
API name changed based on feedback

Reply:
> @vaidyan1-int Query: For one firmware handle could we retrieve multiple type of logs? If yes then we might need to add some logType parameter in https://spec.oneapi.io/level-zero/latest/sysman/api.html#_CPPv425zes_firmware_properties_t, and accordingly we need to pass on additional argument to API which will tell about type of log.

Firmware logs are pure console logs used for debug purposes. They do not differentiate further like OS logs. That's the reason to keep this simple and not planning to add additional enum to identify log type